client/scripts/views/components/sidebar/index.ts

```diff
@@ -6,21 +6,46 @@ import _ from 'lodash';
 import {
   Button, PopoverMenu, MenuItem, Icon, Icons, Tooltip
 } from 'construct-ui';
-
+import produce from 'immer';
 import { selectNode, initChain, navigateToSubpage } from 'app';
 import app from 'state';
 import { ProposalType, ChainBase, ChainNetwork, ChainType } from 'types';
 import { link } from 'helpers';
 import { ChainInfo, NodeInfo } from 'models';
-
 import Moloch from 'controllers/chain/ethereum/moloch/adapter';
 import SubscriptionButton from 'views/components/subscription_button';
 import ChainStatusIndicator from 'views/components/chain_status_indicator';
 import { ChainIcon } from 'views/components/chain_icon';
 import CommunitySelector from 'views/components/sidebar/community_selector';
+import { DiscordIcon, TelegramIcon, ElementIcon, GithubIcon, WebsiteIcon } from '../component_kit/icons';
+import SidebarSection, { SectionGroupProps, SidebarSectionProps, SubSectionProps } from './sidebar_section';
+import { ButtonIntent, FaceliftButton } from '../component_kit/buttons';
 
+// Toggle Tree Definition (3 layers of depth, could add more if desired)
+interface ToggleTree {
```

Comment:
First note is how large this file is. We should absolutely break this up into different files for each "Section" of the sidebar.

Reply:
Around half the file is going to be deleted shortly; it's just being kept right now because another file uses the old sidebar code. But yeah, still probably a good idea.